File: migrations/backend-api/app/models/cliente_formulario.py
"""
Modelo Pivot ClienteFormulario
"""
from sqlalchemy import Column, Integer, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base

class ClienteFormulario(Base):
    __tablename__ = "cliente_formulario"
    
    id = Column(Integer, primary_key=True, index=True)
    cliente_id = Column(Integer, ForeignKey("clientes.id"), nullable=True)
    formulario_id = Column(Integer, ForeignKey("formularios.id"), nullable=True)
    quantidade = Column(Integer, nullable=True)
    ativo = Column(Boolean, default=True)
    deleted_by = Column(Integer, nullable=True)
    
    # Timestamps
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime, nullable=True)  # Soft delete
    
    # Relationships
    cliente = relationship("Cliente", back_populates="cliente_formularios")
    formulario = relationship("Formulario", back_populates="cliente_formularios")

File: php_old_app/app/Models/ClienteFormulario.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class ClienteFormulario extends Model
{
    use SoftDeletes;

    protected $table = 'cliente_formulario';

    protected $fillable = [
        'cliente_id',
        'formulario_id',
        'quantidade',
        'ativo',
        'deleted_by'
    ];

    protected $casts = [
        'ativo' => 'boolean',
        'deleted_by' => 'integer',
        'cliente_id' => 'integer',
        'formulario_id' => 'integer',
        'quantidade' => 'integer',
    ];

    public function cliente()
    {
        return $this->belongsTo(Cliente::class, 'cliente_id');
    }

    public function formulario()
    {
        return $this->belongsTo(Formulario::class, 'formulario_id');
    }

    protected static function boot()
    {
        parent::boot();

        static::addGlobalScope('withRelations', function ($query) {
            $query->with(['cliente', 'formulario']);
        });
    }

}